mobile_verifier/src/speedtests.rs

```diff
@@ -41,6 +53,63 @@ impl Speedtest {
     }
 }
 
+pub struct SpeedtestDaemon {
+    pool: sqlx::Pool<sqlx::Postgres>,
+    config_client: Client,
+    speedtests: Receiver<FileInfoStream<CellSpeedtestIngestReport>>,
+    file_sink: FileSinkClient,
+}
+
+impl SpeedtestDaemon {
+    pub fn new(
+        pool: sqlx::Pool<sqlx::Postgres>,
+        config_client: Client,
+        speedtests: Receiver<FileInfoStream<CellSpeedtestIngestReport>>,
+        file_sink: FileSinkClient,
+    ) -> Self {
+        Self {
+            pool,
+            config_client,
+            speedtests,
+            file_sink,
+        }
+    }
+
+    pub async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
+                    _ = shutdown.clone() => break,
+                    Some(file) = self.speedtests.recv() => {
```

Comment:
as with heartbeats, be good to have the handling of the reports in its own function

Reply:
Addressed 0c7e805